Unable to complete `nix develop` on macos

After installing nix, I cloned the repo and attempted to run `nix develop`. I got the following error message:

```
error: Package ‘mold-1.2.1’ in /nix/store/0cchbcs02ggxbsy3xbmz3dl9vzv5l67w-source/pkgs/development/tools/mold/default.nix:40 is marked as broken, refusing to evaluate.

       a) To temporarily allow broken packages, you can use an environment variable
          for a single invocation of the nix tools.

            $ export NIXPKGS_ALLOW_BROKEN=1

        Note: For `nix shell`, `nix build`, `nix develop` or any other Nix 2.4+
        (Flake) command, `--impure` must be passed in order to read this
        environment variable.

       b) For `nixos-rebuild` you can set
         { nixpkgs.config.allowBroken = true; }
       in configuration.nix to override this.

       c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
         { allowBroken = true; }
       to ~/.config/nixpkgs/config.nix.
(use '--show-trace' to show detailed location information)
```

when I ran `export NIXPKGS_ALLOW_BROKEN=1` and then `nix develop --impure` I get the following error:

```
error: builder for '/nix/store/qinv6i9dxg408i9pj0wwrwy0lnkb2p1s-mold-1.2.1.drv' failed with exit code 2;
       last 10 log lines:
       >       tbb::detail::r1::tbb_exception_ptr::destroy() in libtbb.a(task_group_context.cpp.o)
       >       tbb::detail::r1::tbb_exception_ptr::throw_self() in libtbb.a(task_group_context.cpp.o)
       >       tbb::detail::r1::task_group_context_impl::destroy(tbb::detail::d1::task_group_context&) in libtbb.a(task_group_context.cpp.o)
       >       tbb::detail::r1::context_list::remove(tbb::detail::d1::intrusive_list_node&) in libtbb.a(task_group_context.cpp.o)
       >       tbb::detail::r1::task_group_context_impl::register_with(tbb::detail::d1::task_group_context&, tbb::detail::r1::thread_data*) in libtbb.a(task_group_context.cpp.o)
       >       tbb::detail::r1::task_group_context_impl::bind_to_impl(tbb::detail::d1::task_group_context&, tbb::detail::r1::thread_data*) in libtbb.a(task_group_context.cpp.o)
       >       ...
       > ld: symbol(s) not found for architecture arm64
       > clang-13: error: linker command failed with exit code 1 (use -v to see invocation)
       > make: *** [Makefile:133: mold] Error 1
       For full logs, run 'nix log /nix/store/qinv6i9dxg408i9pj0wwrwy0lnkb2p1s-mold-1.2.1.drv'.
error: 1 dependencies of derivation '/nix/store/qyxsilp10jwbkz5cpg7w7xj5gw20p40l-nix-shell-env.drv' failed to build
```

Total newb so any assistance you can provide will be greatly appreciated.

Env variables:
macOS Monterey 12.5.1
nix (Nix) 2.11.0

--- a/.cargo/config.toml
+++ b/.cargo/config.toml
@@ -1,7 +1,0 @@
-[target.x86_64-unknown-linux-gnu]
-linker = "clang"
-rustflags = ["-C", "link-arg=-fuse-ld=mold"]
-
-[target.aarch64-apple-darwin]
-linker = "clang"
-rustflags = ["-C", "link-arg=-fuse-ld=mold"]

--- a/flake.nix
+++ b/flake.nix
@@ -132,7 +132,6 @@
             perl
             fenix-channel.rustc
             fenix-channel.clippy
-            mold
           ] ++ lib.optionals stdenv.isDarwin [
             libiconv
             darwin.apple_sdk.frameworks.Security

